comments/views.py

```diff
@@ -1,3 +1,84 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from users.utils import IsLoggedIn
+from django.contrib.auth import login, logout
+from rest_framework import permissions, status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.parsers import JSONParser
+from rest_framework.decorators import api_view, renderer_classes
+from rest_framework.renderers import JSONRenderer, TemplateHTMLRenderer
+from django.http import HttpResponse, JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+from users.models import User
+from posts.models import Post
+from .models import Comment
+
+class CreateComment(APIView):
+
+    def post(self,request):
+        try:
+            user=IsLoggedIn(request)
+            if user is None:
+                return Response(status=status.HTTP_401_UNAUTHORIZED)
+            post_id=request.data['pk']
+            content=request.data['content']
+            post=Post.objects.get(pk=post_id)
+            if post is None:
+                return Response(status=status.HTTP_404_NOT_FOUND)
+            try:
+                parent=request.data['parent_id']
+                parent_comment=Comment.objects.get(pk=parent)
+                comment=Comment.objects.filter(user=user,post=post,parent=parent_comment)
+                if len(comment) is not 0:
+                    response={
+                        "message":"comment from this user already exists"
+                    }
+                    return Response(response,status=status.HTTP_400_BAD_REQUEST)
```

Comment:
I think the user will not be able to reply more than once on a same comment. This should not be the case and users should be able to comment as many times as they wants.

Reply:
ok